Location: TestUtils.java

Content: 

public static JobExecutionResult tryExecute(StreamExecutionEnvironment see,String name) throws Exception {

  try {

    return see.execute(name);

  }

 catch (  ProgramInvocationException|JobExecutionException root) {

    Throwable cause=root.getCause();

    int depth=0;

    while (!(cause instanceof SuccessException)) {

      if (cause == null || depth++ == 20) {

        root.printStackTrace();

        fail("Test failed: " + root.getMessage());

      }

 else {

        cause=cause.getCause();

      }

    }

  }

  return null;

}

Location: TestUtils.java

Content: 

public static JobExecutionResult tryExecute(StreamExecutionEnvironment see,String name) throws Exception {

  try {

    return see.execute(name);

  }

 catch (  ProgramInvocationException|JobExecutionException root) {

    Throwable cause=root.getCause();

    int depth=0;

    while (!(cause instanceof SuccessException)) {

      if (cause == null || depth++ == 20) {

        root.printStackTrace();

        fail("Test failed: " + root.getMessage());

      }

 else {

        cause=cause.getCause();

      }

    }

  }

  return null;

}

